cloud/src/meta-service/meta_service_resource.cpp

```diff
@@ -308,6 +308,61 @@ static std::string next_available_vault_id(const InstanceInfoPB& instance) {
     return std::to_string(prev + 1);
 }
 
+namespace detail {
+
+// Removes any trailing `c` in `str`
+void strip_trailing(std::string& str, char c) {
+    size_t end = str.find_last_not_of(c);
+    if (end == std::string::npos) {
+        str = "";
+    } else {
+        str.resize(end + 1);
+    }
+}
+
+// Removes any leading `c` in `str`
+void strip_leading(std::string& str, char c) {
+    size_t start = str.find_first_not_of(c);
+    if (start == std::string::npos) {
+        str = "";
+    } else if (start > 0) {
+        str = str.substr(start);
+    }
+}
+
+// Validate and normalize hdfs prefix. Return true if prefix is valid.
+bool normalize_hdfs_prefix(std::string& prefix) {
```

Comment:
Add comment that, if the input is invalid, it is not touched after calling this function.